auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthContext.java

```diff
@@ -0,0 +1,21 @@
+package com.vaadin.auth.starter;
+
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+import java.util.Optional;
+
+/**
+ * An interface to access the authentication context of the application
+ *
+ * @author Vaadin Ltd
+ * @since 1.0
+ */
+public interface VaadinAuthContext {
+    /**
+     * Gets the authenticated user on the application done by the SSO providers
+     *
+     * @return an optional {@link OidcUser} object if the user is authenticated
+     *         or an optional {@code null} otherwise
+     */
```

Comment:
Better to clarify also here what to expect from the method, e.g. empty if the user is authenticated by other means (not ODIC).
```suggestion

    /**
     * Gets an {@link Optional} with an instance of the current user if it has
     * been authenticated by a OpenID Connect provider, or empty if the user is
     * not authenticated or has been authenticated by other means.
     *
     * @return an {@link Optional} with the current OIDC authenticated user, or
     *         empty if none available
     */
```